scipy/integrate/_bvp.py

```diff
@@ -1078,12 +1085,16 @@ def solve_bvp(fun, bc, x, y, p=None, S=None, fun_jac=None, bc_jac=None,
 
         col_res, y_middle, f, f_middle = collocation_fun(fun_wrapped, y,
                                                          p, x, h)
+        bc_res = bc_wrapped(y[:, 0], y[:, -1], p)
+        max_bc_res = np.max(abs(bc_res)) * tol / tol_bc
+
         # This relation is not trivial, but can be verified.
         r_middle = 1.5 * col_res / h
         sol = create_spline(y, f, x, h)
         rms_res = estimate_rms_residuals(fun_wrapped, sol, x, h, p,
                                          r_middle, f_middle)
         max_rms_res = np.max(rms_res)
+        max_res = np.max([max_rms_res, max_bc_res])
```

Comment:
I think it's better to consider two residuals separately.

Reply:
`max_res` was removed and the two residuals are now considered and reported separately.